Content.Server/Medical/DefibrillatorSystem.cs

```diff
@@ -1,151 +1,43 @@
-            return;
-        args.Handled = TryStartZap(uid, target, args.User, component);
-    }
-
-    private void OnDoAfter(EntityUid uid, DefibrillatorComponent component, DefibrillatorZapDoAfterEvent args)
-    {
-        if (args.Handled || args.Cancelled)
-            return;
-
-        if (args.Target is not { } target)
-            return;
-
-        if (!CanZap(uid, target, args.User, component))
-            return;
-
-        args.Handled = true;
-        Zap(uid, target, args.User, component);
-    }
-
-    public bool CanZap(EntityUid uid, EntityUid target, EntityUid? user = null, DefibrillatorComponent? component = null)
-    {
-        if (!Resolve(uid, ref component))
-            return false;
-
-        if (!_toggle.IsActivated(uid))
-        {
-            if (user != null)
-                _popup.PopupEntity(Loc.GetString("defibrillator-not-on"), uid, user.Value);
-            return false;
-        }
-
-        if (_timing.CurTime < component.NextZapTime)
-            return false;
-
-        if (!TryComp<MobStateComponent>(target, out var mobState))
-            return false;
-
-        if (!_powerCell.HasActivatableCharge(uid, user: user))
-            return false;
-
-        if (_mobState.IsAlive(target, mobState))
-            return false;
-
-        if (!component.CanDefibCrit && _mobState.IsCritical(target, mobState))
-            return false;
-
-        return true;
-    }
-
-    public bool TryStartZap(EntityUid uid, EntityUid target, EntityUid user, DefibrillatorComponent? component = null)
-    {
-        if (!Resolve(uid, ref component))
-            return false;
-
-        if (!CanZap(uid, target, user, component))
-            return false;
-
-        _audio.PlayPvs(component.ChargeSound, uid);
-        return _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, user, component.DoAfterDuration, new DefibrillatorZapDoAfterEvent(),
-            uid, target, uid)
-            {
-                BlockDuplicate = true,
-                BreakOnHandChange = true,
-                NeedHand = true,
-                BreakOnMove = !component.AllowDoAfterMovement
-            });
-    }
-
-    public void Zap(EntityUid uid, EntityUid target, EntityUid user, DefibrillatorComponent? component = null, MobStateComponent? mob = null, MobThresholdsComponent? thresholds = null)
+    public override void Zap(EntityUid uid, EntityUid target, EntityUid user, DefibrillatorComponent component, MobStateComponent? mob = null)
     {
-        if (!Resolve(uid, ref component) || !Resolve(target, ref mob, ref thresholds, false))
-            return;
-
-        // clowns zap themselves
-        if (HasComp<ClumsyComponent>(user) && user != target)
-        {
-            Zap(uid, user, user, component);
-            return;
-        }
-
+        // TODO : powercell should be rewritten to shared instead of strictly be on Server side
         if (!_powerCell.TryUseActivatableCharge(uid, user: user))
             return;
```

Comment:
This method doesn't call the Shared.Zap. You'll need to call `base.Zap` or something here and maybe add a bool return value for if the shared code returns.

Reply:
fixed by calling base.Zap in server's overrided Zap